test/micro/org/openjdk/bench/vm/runtime/NMTAnalyze.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class NMTAnalyze {
+  public static class BenchmarkRecord {
+    public String jmh_method;
+    public int N, threads, cnt;
+    public double score, error;
+    public String mode;
+    public String units;
+  }
+  public static class BenchmarkFields {
+    static int NMT_METHOD    = 0;
+    static int N             = 1;
+    static int THREADS       = 2;
+    static int MODE          = 3;
+    static int CNT           = 4;
+    static int SCORE         = 5;
+    static int QUESTION_MARK = 6;
+    static int ERROR         = 7;
+    static int UNITS         = 8;
+  }
+  public static BufferedReader reader;
+  public static String fileName;
+  public static List<BenchmarkRecord> records = new ArrayList<>();
+  private static void skipUntilReport() {
+    try {
+      String line = reader.readLine();
+      while (line != null) {
+        String ReportStartKeyword = "Benchmark";
+        if (!line.startsWith(ReportStartKeyword) || line.contains("#")) {
```

Comment:
Are there lines which starts with `Benchmark` but contains a `#` that we need to skip?

Reply:
Yes, there are. for each run there is a part like this. with `# Benchmark`:
```
# JMH version: 1.37
# VM version: JDK 22-internal, OpenJDK 64-Bit Server VM,......in...
# VM invoker:.....
# VM options: -XX:NativeMemoryTracking=detail --add-exports java.base/jdk.internal.misc=ALL-UNNAMED -Djava.library.path=...
# Blackhole mode: compiler (auto-detected, use -Djmh.blackhole.autoDetect=false to disable)
# Warmup: 2 iterations, 1 s each
# Measurement: 5 iterations, 1 s each
# Timeout: 10 min per iteration
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Average time, time/op
# Benchmark: org.openjdk.bench.vm.runtime.NMTBenchmark.NMTDetail.mixAallocateFreeMemory
# Parameters: (N = 100000, THREADS = 4)
```